packages/gatsby-remark-images/src/index.js

```diff
@@ -83,7 +83,7 @@ module.exports = (
     const getCaptionString = () => {
       const captionOptions = Array.isArray(options.showCaptions)
         ? options.showCaptions
-        : options.showCaptions === true
+        : options.showCaptions === true && node.alt !== EMPTY_ALT
```

Comment:
This line introduces an unwanted behavior for the generation of image title. Actually, the logic to detect `EMPTY_ALT` should be put after line 98.